### PR TITLE
[PodspecSource] Evaluate :podspec dependencies in their own directory if the path is local.

### PR DESCRIPTION
**Pull request by segiddins**
This fixes an issue @tgoyne has been having.

@kylef, a quick merge of this would be much appreciated. The failures are in the integration spec, and are expected.

### CODE REVIEW

File: lib/cocoapods/external_sources/podspec_source.rb [hunk 1]
```diff
@@ -9,13 +9,18 @@ class PodspecSource < AbstractExternalSource
...
-          begin
-            open(podspec_uri) { |io| store_podspec(sandbox, io.read, is_json) }
-          rescue OpenURI::HTTPError => e
-            status = e.io.status.join(' ')
-            raise Informative, "Failed to fetch podspec for `#{name}` at `#{podspec_uri}`.\n Error: #{status}"
+          podspec_path = Pathname(podspec_uri)
+          if podspec_path.exist?
+            store_podspec(sandbox, podspec_path, podspec_path.extname == '.json')
+          else
+            is_json = podspec_uri.split('.').last == 'json'
```
Comment (kylef) [inline lib/cocoapods/external_sources/podspec_source.rb]: What's the purpose of using two different pieces of logic to determine if the file is a JSON podspec?

```ruby
podspec_path.extname == '.json'
podspec_uri.split('.').last == 'json'
```

Comment (segiddins) [inline lib/cocoapods/external_sources/podspec_source.rb]: because `podspec_path` is a `Pathname` and `podspec_uri` is only a `String`

Comment (kylef) [inline lib/cocoapods/external_sources/podspec_source.rb]: The pathname is created before and can still be used at this point.

Comment (segiddins) [inline lib/cocoapods/external_sources/podspec_source.rb]: I suppose...

Comment (segiddins) [inline lib/cocoapods/external_sources/podspec_source.rb]: Done

**Comment by kylef** [comment]
@segiddins Good to merge once rebased.

**Comment by fatuhoku** [comment]
I'm suspecting this PR might resolve #3070?

**Pull request merged**